source/slang/slang-ir-diff-jvp.cpp

```diff
@@ -352,16 +389,38 @@ struct JVPTranscriber
     }
 
     // Logic for whether a primal instruction needs to be replicated
-    // in the differential function. For puerly functional blocks with
-    // no side-effects, it's safe to replicate everything except the
-    // return instruction.
-    //
+    // in the differential function. We detect and avoid replicating 
+    // side-effect instructions.
+    // 
     bool requiresPrimalClone(IRBuilder*, IRInst* instP)
     {
         if (as<IRReturn>(instP))
             return false;
-        else
-            return true;
+        else if (auto paramP = as<IRParam>(instP))
```

Comment:
For naming conventions, I don't know what does the `P` mean in `paramP`, or `instD` etc. in your code. We normally don't use single-letter abbrevations since they are confusing. I think just `param` here would be sufficient.

Reply:
I'm using 'P' to represent the _primal_ (or 'original') instructions that are computing `f(x)`. 'D' postfix is for instructions that are computing the corresponding differential (i.e. for df(x, dx)) for that instruction.
Yes, this could be confusing. I could switch to full postfixes: `instPrimal` and `instDiff`/`instDifferential` but it might cause the code to look bloated. Alternatively `inst` and `diffInst` maybe?

Reply:
Yeah that looks good, or `originalInst` and `diffInst` to be more explicit.